components_library/boards/lupv_board.py

```diff
@@ -320,21 +362,39 @@ def generate_device_tree(self, outdir: str) -> None:
         int_state = FdtState(addr_cells=0, interrupt_cells=1)
         plic_node.append(int_state.addrCellsProperty())
         plic_node.append(int_state.interruptCellsProperty())
-        phandle = int_state.phandle(plic)
+        phandle = state.phandle(plic)
         plic_node.append(FdtPropertyWords("phandle", [phandle]))
-        plic_node.append(FdtPropertyWords("riscv,ndev", [self.pic.n_src - 1]))
+        plic_node.append(FdtPropertyWords("riscv,ndev", 0))
```

Comment:
Was this wrong before? Can you explain this change a bit? (I mostly just want to understand the device tree better)

Reply:
To my understanding, we set it as 0 here because now the LupIO-PIC handles all the devices, so the Plic can be set to 0.  However, we made this change because it was one of the steps in the README we followed.

Reply:
I see! Thanks!

Reply:
Also, BBL can recognize when `riscv,ndev==0` and it returns from PLIC-related functions faster.